### PR TITLE
loggerd: split bootlog to a separate program

### PR DESCRIPTION
**Pull request by deanlee**

### CODE REVIEW

File: selfdrive/loggerd/logger.cc [hunk 1]
```diff
@@ -12,12 +12,102 @@
...
 #include "logger.h"
 
+void append_property(const char* key, const char* value, void *cookie) {
+  std::vector<std::pair<std::string, std::string> > *properties =
+    (std::vector<std::pair<std::string, std::string> > *)cookie;
+
+  properties->push_back(std::make_pair(std::string(key), std::string(value)));
+}
+
+static bool file_exists(const std::string& fn) {
```
Comment (adeebshihadeh) [inline selfdrive/loggerd/logger.cc:36]: let's move this to util

File: selfdrive/loggerd/SConscript [hunk 2]
```diff
@@ -1,7 +1,9 @@
 Import('env', 'arch', 'cereal', 'messaging', 'common', 'visionipc', 'gpucommon')
 
-src = ['loggerd.cc', 'logger.cc']
-libs = ['zmq', 'capnp', 'kj', 'z',
+
+logger_lib = env.Library('logger', ["logger.cc"])
+src = ['loggerd.cc']
```
Comment (adeebshihadeh) [inline selfdrive/loggerd/SConscript:5]: just put this directly in the env.Program call

File: selfdrive/loggerd/logger.cc [hunk 3]
```diff
@@ -12,12 +12,97 @@
...
 #include "logger.h"
 
+void append_property(const char* key, const char* value, void *cookie) {
+  std::vector<std::pair<std::string, std::string> > *properties =
+    (std::vector<std::pair<std::string, std::string> > *)cookie;
+
+  properties->push_back(std::make_pair(std::string(key), std::string(value)));
+}
+
+kj::Array<capnp::word> gen_init_data() {
```
Comment (adeebshihadeh) [inline selfdrive/loggerd/logger.cc:36]: let's make this log_gen_init_data to match the pattern we use for sentinel

File: selfdrive/loggerd/SConscript (hunk 2 above)
Comment (deanlee) [reply selfdrive/loggerd/SConscript]: no,we can't

**Comment by adeebshihadeh** [comment]
Nice, this should make loggerd.cc a lot more readable! Can you add bootlog to the gitignore in the openpilot root? We should have a CI test for that.

File: selfdrive/loggerd/SConscript (hunk 2 above)
Comment (adeebshihadeh) [reply selfdrive/loggerd/SConscript:5]: why not?

File: selfdrive/loggerd/logger.cc (hunk 3 above)
Comment (deanlee) [reply selfdrive/loggerd/logger.cc]: call `log_init_data` in `logger_init` now.Is this what you want

File: selfdrive/loggerd/SConscript (hunk 2 above)
Comment (deanlee) [reply selfdrive/loggerd/SConscript:5]: `src` is modified by the following conditional statement

Comment (adeebshihadeh) [reply selfdrive/loggerd/SConscript:5]: ah, didn't see it in the diff view :)

File: selfdrive/loggerd/logger.h [hunk 4]
```diff
@@ -5,7 +5,13 @@
 #include <stdint.h>
 #include <pthread.h>
 #include <bzlib.h>
-
+#include <kj/array.h>
+#include <capnp/serialize.h>
+#if defined(QCOM) || defined(QCOM2)
```
Comment (adeebshihadeh) [inline selfdrive/loggerd/logger.h:10]: add some whitespace here

Comment (adeebshihadeh) [reply selfdrive/loggerd/logger.h:10]: let's also just make logger.c into a cc file so we can remove the extern c stuff

File: selfdrive/loggerd/loggerd.cc [hunk 5]
```diff
@@ -495,11 +362,7 @@ int main(int argc, char** argv) {
   }
 
   // init logger
-  {
-    auto words = gen_init_data();
-    auto bytes = words.asBytes();
-    logger_init(&s.logger, "rlog", bytes.begin(), bytes.size(), true);
-  }
+  logger_init(&s.logger, "rlog", true);
```
Comment (adeebshihadeh) [inline selfdrive/loggerd/loggerd.cc:365]: this is so much nicer :)

File: selfdrive/loggerd/logger.cc (hunk 3 above)
Comment (adeebshihadeh) [reply selfdrive/loggerd/logger.cc]: yes

File: selfdrive/loggerd/loggerd.cc (hunk 5 above)
Comment (deanlee) [reply selfdrive/loggerd/loggerd.cc:365]: I think so :laughing:

**Pull request merged**